src/types.rs

```diff
@@ -9,6 +9,25 @@ pub struct Game {
     pub history: Vec<World>,
 }
 
+impl Game {
```

Comment:
`impl Game` could be in another file from the struct for organization sake, but this is fine for now 👍